src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java

```diff
@@ -46,8 +57,47 @@ public void check(Network network) throws NetworkModificationException {
         if (network.getGenerator(modificationInfos.getEquipmentId()) != null) {
             throw new NetworkModificationException(GENERATOR_ALREADY_EXISTS, modificationInfos.getEquipmentId());
         }
+
+        // check connectivity
         ModificationUtils.getInstance().controlConnectivity(network, modificationInfos.getVoltageLevelId(),
                 modificationInfos.getBusOrBusbarSectionId(), modificationInfos.getConnectionPosition());
+
+        // check min max reactive limits
+        if (modificationInfos.getMinimumReactivePower() != null && modificationInfos.getMaximumReactivePower() != null) {
+            if (Double.isNaN(modificationInfos.getMinimumReactivePower())) {
+                throw new NetworkModificationException(CREATE_GENERATOR_ERROR, "Generator '" + modificationInfos.getEquipmentId() + "' : minimum reactive power is not set");
```

Comment:
What about defining
```
    private static NetworkModificationException makeGeneratorException(String generatorId, String msgSuffix) {
        return new NetworkModificationException(CREATE_GENERATOR_ERROR, "Generator '" + generatorId + msgSuffix);
}
```
and use it like
```
                throw makeGeneratorException(modificationInfos.getEquipmentId(), "' : minimum reactive power is not set");
```
?

Reply:
Done